modules/core/src/lib/widget-manager.ts

```diff
@@ -57,61 +63,119 @@ const ROOT_CONTAINER_ID = '__root';
 export class WidgetManager {
   deck: Deck;
   parentElement?: HTMLElement | null;
-  containers: {[id: string]: HTMLDivElement} = {};
-  widgets: Widget[] = [];
-  lastViewports: {[id: string]: Viewport} = {};
+
+  /** Widgets added via the imperative API */
+  private defaultWidgets: Widget[] = [];
+  /** Widgets received from the declarative API */
+  private widgets: Widget[] = [];
+  /** Resolved widgets from both imperative and declarative APIs */
+  private resolvedWidgets: Widget[] = [];
+  private containers: {[id: string]: HTMLDivElement} = {};
+  private lastViewports: {[id: string]: Viewport} = {};
```

Comment:
Is lastViewports related to widgets or should we separate it with an empty line?

Reply:
It's related to containers, so I'll group those two and add JSDocs